rest/utilities_testing.go

```diff
@@ -814,3 +1046,95 @@ func EnableBlipSyncLogs() {
 	base.EnableLogKey("Sync")
 	base.EnableLogKey("Sync+")
 }
+
+
+type AttachmentMap map[string]*DocAttachment
+
+type DocAttachment struct {
+	ContentType string `json:"content_type,omitempty"`
+	Digest      string `json:"digest,omitempty"`
+	Length      int    `json:"length,omitempty"`
+	Revpos      int    `json:"revpos,omitempty"`
+	Stub        bool   `json:"stub,omitempty"`
+	Data        []byte `json:-` // tell json marshal/unmarshal to ignore this field
+}
+
+// --------------------------------------------
+
+type RestDocument map[string]interface{}
```

Comment:
Why not define as a struct w/ id, rev, attachments?  Is there a reason to preserve the rest of the doc's properties when unmarshalling as RestDocument?

Reply:
We want to move away from sticking things like _id and _rev in the document body.  Are we making that chabge more difficult in future by defining the tests based on RestDocument?

Reply:
> Is there a reason to preserve the rest of the doc's properties when unmarshalling as RestDocument?

It was mainly because I wanted `GetAllDocsViaChanges()` aka `PullDocs()` to return the entire docs, but with convenience methods to get special properties and attachment structs that could be inspected.  Technically though, for the purposes of testing, I think getting the entire docs wasn't actually needed.  I guess I was thinking forward for other tests that might need to inspect the doc bodies for some reason.

I could go ahead and change it to a struct with named fields for now, since that's simpler to understand, however then the code wouldn't be as generally useful .. eg, if it were being used in a blip-sync client library that a CLI was using, which wanted to pull all of the latest doc revs and wanted to emit them to stdout or to files .. then it would only have the special fields and the rest of the doc bodies would be discarded.   This might need some more discussion.

> We want to move away from sticking things like _id and _rev in the document body. Are we making that change more difficult in future by defining the tests based on RestDocument?

Possibly, but I think if we make that change, we're going to have to change a ton of code and tests anyway.  It might be tricky to start "future proofing" this too early, but we can catch up more on that though.

Reply:
My point was that RestDocument looks like something that should have been included in non-test code years ago, but wasn't.  This doesn't mean it would be a bad idea to have it in the test code, but at some point we'll need to bite the bullet and get this dealt with in the non-test code.